Add compliance wrapper and script new T-REX flow

Thaleszh opened the pull request:
# Description

Creates the flow for a new T-REX including a liquidity pool midway.
Also adds a compliance wrapper that allows whitelisting an address. This way we can use token pairs to do transfers.

## Type of change


- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation (updates documentation on the project)
- [x] test (For updates on tests)

# How Has This Been Tested?

Tested manually

Commit:
feat: add new compliance wrapper

Commit:
feat: add incomplete new trex flow

Commit:
feat: deploy compliance wrapper

Commit:
fix: add script to remove old compliance and update with compliance wrapper

Commit:
feat: add new t-rex flow

Commit:
fix: get deploy working

Commit:
chore: run cargo fmt

jhelison commented on contracts/compliance-wrapper/src/contract.rs:
```diff
@@ -0,0 +1,363 @@
+    use cosmwasm_std::{to_json_binary, Addr, QueryRequest, WasmQuery};
+    use utils::owner_roles::{IsOwnerResponse, QueryMsg};
+
+    pub fn check_role(deps: Deps, owner: Addr, role: OwnerRole) -> Result<(), ContractError> {
```
Missing docstring

Commit:
fix: rollback change to test

jhelison commented on contracts/compliance-wrapper/src/contract.rs:
```diff
@@ -0,0 +1,379 @@
+        address: Addr,
+    ) -> Result<Response, ContractError> {
+        WHITELISTED_ADDRESSES.remove(deps.storage, address);
+        Ok(Response::new().add_attribute("action", "remove_address_from_whitelist"))
```
Missing docstrings

jhelison commented on contracts/compliance-wrapper/src/contract.rs:
```diff
@@ -0,0 +1,379 @@
+            amount,
+        )?),
+    }
+}
```
Missing docstring

jhelison commented on contracts/compliance-wrapper/src/contract.rs:
```diff
@@ -0,0 +1,379 @@
+        amount: Option<Uint128>,
+    ) -> StdResult<bool> {
+        // Get compliance module and whitelisted addresses
+        let module_address = COMPLIANCE_MODULE_ADDRESS.load(deps.storage)?;
+
+        // Replace whitelisted addrs with None
+        let from = if is_whitelisted(deps, from.clone()) {
+            None
+        } else {
+            from
+        };
+        let to = if is_whitelisted(deps, to.clone()) {
+            None
+        } else {
```
This is interesting, good job. 100% sure that current compliance modules handle well from and to as None?

Thaleszh replied:
Yep, I checked them both. They were already ready to handle calls that were missing one or the other and checked it individually.

jhelison commented on contracts/compliance-wrapper/src/contract.rs:
```diff
@@ -0,0 +1,379 @@
+        amount: Option<Uint128>,
+    ) -> StdResult<bool> {
+        // Get compliance module and whitelisted addresses
+        let module_address = COMPLIANCE_MODULE_ADDRESS.load(deps.storage)?;
+
+        // Replace whitelisted addrs with None
+        let from = if is_whitelisted(deps, from.clone()) {
+            None
+        } else {
+            from
+        };
+        let to = if is_whitelisted(deps, to.clone()) {
+            None
+        } else {
```
Da heck is going on here? Are we not setting the option above on line 171?

Thaleszh replied:
Oh shit, I must have gotten something in my head mid implementation. Yes, it is duplicated.

Thaleszh replied:
I removed it, it doesn't change how it works but for a future deploy it won't have that

jhelison reviewed (commented):
We should one of:
- The word whitelist on the contract name
Or/And
- Have simple docs
This wrapper stuff can be quite confusing until people check the states and the execution

Commit:
fix: remove duplicated is_whitelisted check for from

Commit:
docs: add missing docstrings

Commit:
chore: rename contract

Commit:
docs: add a simple readme

Thaleszh commented:
> We should one of:
> 
>     * The word whitelist on the contract name
>       Or/And
> 
>     * Have simple docs
>       This wrapper stuff can be quite confusing until people check the states and the execution

I renamed the contract and added a simple readme.
